src/component.mojo

```
@@ -116,84 +127,55 @@ struct ComponentManager:
-
-        Raises:
-            Error: If check_existent and the component type has already been registered.
-            Error: If the maximum number of components has been reached.
+            T: The component type.
         """
 
         @parameter
-        if check_existent:
-            if T.get_type_identifier() in self._components:
-                raise Error(
-                    "A component with hash "
-                    + str(T.get_type_identifier())
-                    + " has already been registered."
-                )
-
-        if len(self._components) >= self.max_size:
-            raise Error(
-                "We cannot register more than "
-                + str(self.max_size)
-                + " elements in a component manager of this type."
-            )
-
-        component_info = ComponentInfo.new[T](Self.Id(len(self._components)))
-        self._components[T.get_type_identifier()] = component_info
-
-    fn get_id[T: ComponentType](inout self) raises -> Self.Id:
-        """Get the ID of a component type.
+        for i in range(len(VariadicList(component_types))):
 
-        If the component does not yet have an ID, register the component.
+            @parameter
+            if _type_is_eq[T, component_types[i]]():
+                return i
 
-        Parameters:
-            T: The component type.
+        # This constraint will fail if the component type is not in the list.
+        constrained[
+            _contains_type[T, *component_types](),
+            "The used component is not in the component parameter list.",
+        ]()
 
-        Raises:
-            Error: If the component was not registered and the maximum number of components has been reached.
-        """
-        if T.get_type_identifier() in self._components:
-            return self._components[T.get_type_identifier()].id
-        else:
-            return self._register[T, False]().id
+        # This is unreachable.
+        return -1
```

Maybe raise an error here?

---

Then I would need to make the function raising. For an unreachable error, this would be a disproportionate cost.

---

Would you strictly need to do that? I just mean, because this is not an exception that is intended to be caught. So I would panic, but there is no such thing in Mojo.

---

Yeah, and it would not provide a stack trace. So `raise` without `raises` is not possible?

---

No, raise requires raises. This should be literally unreachable, as the compile time statement _contains_type does exactly the same thing as the for loop. I would have put a false constraint, but the compiler is not smart enough to ignore constraints after a return.